### PR TITLE
Fix all ReSpec warnings

### PR DESCRIPTION
**Pull request by alexshalamov**
This PR fixes ReSpec warnings and improves cross-linking inside document.

### CODE REVIEW

**Comment by alexshalamov** [comment]
@marcoscaceres please take a look.

File: index.html [hunk 1]
```diff
@@ -236,6 +236,13 @@
           };
         </pre>
         <p>
+          The <dfn>ScreenOrientation</dfn> interface provides functionality to
+          <code><a data-link-for='ScreenOrientation' data-lt='lock'>lock()</a>
+          </code> and <code><a data-link-for='ScreenOrientation' data-lt=
+          'unlock'>unlock()</a></code> screen orientation, as well as to get
+          the orientation information about the <a>Screen</a> object.
+        </p>
+        <p>
```
Comment (mounirlamouri) [inline index.html]: Why did you add this?

Comment (alexshalamov) [reply index.html]: Because definition for `ScreenOrientation` interface was missing.

Comment (alexshalamov) [reply index.html]: I can remove this section and add `<dfn>` to the header of the section.

Comment (mounirlamouri) [reply index.html]: I haven't used respec in a while but maybe you could simply replace the `<a>ScreenOrientation</a>` with `<code>ScreenOrientation</code>`?

Comment (alexshalamov) [reply index.html]: Fixed. Thanks

**Review by mounirlamouri — APPROVED** [review]

**Comment by mounirlamouri** [comment]
Thanks! :)

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html]: No, don't ever use code :( Only use`<a>` or it won't auto link.

Comment (mounirlamouri) [reply index.html]: dfn ended up being used but in the title instead of adding a new paragraph. Though, I wonder why the interface in the WebIDL snippet wasn't linkified.

File: index.html [hunk 2]
```diff
@@ -223,7 +223,7 @@
       </section>
       <section>
         <h2>
-          <code>ScreenOrientation</code> interface
+          <code><dfn>ScreenOrientation</dfn></code> interface
```
Comment (marcoscaceres) [inline index.html:226]: you don't need the code elements here.

File: index.html [hunk 3]
```diff
@@ -248,7 +248,7 @@
           <a>responsible document</a>'s <a>default orientation</a>.
         </p>
         <p class='note'>
-          <a for='ScreenOrientation' data-lt='unlock'>unlock()</a> does not
+          <a data-link-for='ScreenOrientation' data-lt='unlock'>unlock()</a> does not
```
Comment (marcoscaceres) [inline index.html:251]: we should move `data-link` up to the section element.

File: index.html [hunk 4]
```diff
@@ -274,7 +274,7 @@
       </section>
       <section>
         <h2>
-          <code>OrientationType</code> enum
+          <code><dfn>OrientationType</dfn></code> enum
```
Comment (marcoscaceres) [inline index.html:277]: same here about `code`

Comment (marcoscaceres) [reply index.html:277]: ... and globally.

File: index.html [hunk 5]
```diff
@@ -360,7 +360,7 @@
         <ol>
           <li>If the screen width is greater than the screen height, set the
           <a>document</a>'s <a>current orientation type</a> to
-            <code><dfn for='OrientationTypelandscape'>landscape-primary</dfn></code>
+            <code><dfn for='OrientationType'>landscape-primary</dfn></code>
```
Comment (marcoscaceres) [inline index.html:363]: please remove the for attribute, as it's not supported.

File: index.html [hunk 6]
```diff
@@ -662,12 +662,11 @@
                 </ol>
               </li>
               <li>If the orientation change was triggered by a user gesture such
-              as the user turning the device, as opposed to a call to <a for=
-              'ScreenOrientation'>lock</a>, the <a>task</a> MUST be annotated
-              with <code>process user orientation change</code> when running the
-              next step.
+              as the user turning the device, as opposed to a call to
+              <a data-link-for='ScreenOrientation' data-lt='lock'>lock</a>, the
```
Comment (marcoscaceres) [inline index.html:666]: this will auto link once you move `data-link-for` up to the section element.

File: index.html (hunk 2 above)
Comment (alexshalamov) [reply index.html:226]: @marcoscaceres I tried not to remove anything that was already in the spec. Should I do it in follow-up PR or you want to revert this one?

File: index.html (hunk 4 above)
Comment (alexshalamov) [reply index.html:277]: @marcoscaceres Thanks for the hint, can I make new PR for these fixes?

File: index.html (hunk 2 above)
Comment (marcoscaceres) [reply index.html:226]: A new new PR would be great. Really appreciate your time and help here.
